### PR TITLE
DOC: Reword description about patching in preview mode

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR rewords the documentation about patching in preview mode to make it less ambiguous.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

### CODE REVIEW

File: doc/sources/preview.rst [hunk 1]
```diff
@@ -29,8 +29,9 @@ for being enabled by default for all users:
 This type of functionality is available under **preview mode** of |sklearnex| and located in
 the corresponding module (`sklearnex.preview`).
 
-Preview functionality *may* or *may not* participate in patching of Scikit-learn.
-For example, a preview estimator may be a replacement for a stock one or a completely new one.
+Functionalities under preview will be made available after patching when preview mode is enabled,
```
Comment (icfaust) [inline doc/sources/preview.rst:33]: Are we sure that the sklearnex-only estimators don't get patched-in? https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/sklearnex/dispatcher.py#L62  It takes the mapping without preview, and then adds the preview estimators on top, and for the sklearnex-only estimators, if it didn't patch in, we would not have sklearn conformance deselections for those estimators (mostly for ```test_common.py```). Maybe I misunderstood? Let me know.  This PR should be quick to approve once we iron this out (figure out what I got wrong here).

Comment (david-cortes-intel) [reply doc/sources/preview.rst:33]: Yes, looks like patching still makes them importable from `sklearn`. Modified the docs here and in the section about extension estimators.
